### ISSUE TITLE
Music in ogg format not playing

### ISSUE THREAD

**Issue by darkstar252** [shinyquagsire23/OpenJKDF2#342]
It seems like the program is attempting at least to play music but no music is comming from the speakers.
stdMci: Set vol 1.000000
stdMci: play track 2 to 3
stdMci: Playing music `MUSIC/1/Track2.ogg'
stdMci: Set vol 0.000000

**Comment by darkstar252** [comment]
Corrected on the game on the first level on the bar there is some music that can be ear with the original game whichi i think it was in the assets of the original game. And when you exited the game the level 1 cd track is playing. Another thing is when you restart the level the music doesn't restart from the beggining. And it sound at very low volume even at maximum in the configuration.

**Comment by shinyquagsire23** [comment]
Oh right, on level 1 the soundtrack will fade in after you exit the bar, this is intentional scripted behavior which CD-less releases failed to emulate (The game itself sets the volume to 0.0 and then ramps it up, you'll see it in the console).

I might need to double-check on the level restart thing though, I'm not totally sure that the original release did that.

**Comment by darkstar252** [comment]
But the cantina music isnt ear from my headphones and that confuses me a lot , to be honest. An i appreciate your effort and the effort of the team. 
Thanks :)

**Comment by shinyquagsire23** [comment]
Oh, yeah the cantina music should definitely be there, might double-check that then. I will say, that music I believe is controlled by the sfx volume since it's environmental/in-world and not soundtrack.

I'll also see if there's something odd with the level restart (I wonder if it depended on the CD drive). I kinda eyeballed that subsystem when retrofitting it for OGG so maybe I missed some behavior.

**Comment by darkstar252** [comment]
Hey Max.
Something new about this? At least to close the bug or try to compile another release if you view all of those things.

Thank you very much.

**Comment by shinyquagsire23** [comment]
Yeah the main holdup on another release is mostly just fixing build issues, the last release had problems where a C++ redistributable was required when I tried to avoid that.

**Comment by darkstar252** [comment]
There is not relevant to this bug , but i can compile it on windows xp 32 bit sp3?? Just for the sake to make a binary for windows xp and thank you very much again tell me if can close this , when it's ready even if i need to compile on my linux distro again. :)

**Comment by shinyquagsire23** [comment]
If you try hard enough, maybe? I've never tested it and it does require a fairly recent CMake version, but tbh I've considered making a 32-bit build w/ OpenGL 2 and an XP target.

**Comment by darkstar252** [comment]
Of course when i send those messages i am using linux i have at least 10 computers in my house(i know that is completely crazy) even i have a pentium 4 (socket 775) and an athlon 64 3000+(socket 754) and the usually modern ones.

**Comment by shinyquagsire23** [comment]
So, I looked into it, the soundtrack gets invoked from a `startup` COG script event. The problem though is that mission restarts don't actually reinvoke all of the `startup` events, they make a quicksave right after the level is loaded and load that instead I guess? So I don't think it actually would restart the soundtrack on level restarts (maybe this is another one of those things that MoTS fixed though... I keep finding little changes they made here and there)

**Comment by shinyquagsire23** [comment]
Hmmm there's hints that the menu may have paused the soundtrack at some point, maybe
https://github.com/shinyquagsire23/OpenJKDF2/commit/55e3344a5669c22dafffa22fa91139ea3e6fb362#diff-05e81798c21cd56cfdb40bd52d5e8082d5bb61b46cdd5d4cb9d06c06cc845de1R165

**Comment by darkstar252** [comment]
It seems to work, but for any strange reason on first level the OGG Music(the audio cd music) begins when you leave the cantina. :) Not when entering level one :) . And yes i think not the main menu but the game menu paused the soundtrack from original CD (at least is the best behavior)

**Comment by darkstar252** [comment]
If you need for me to test this thing on the original game , tell me. I install the game on other machines , i had with windows xp or windows vista(because i own the original game from back them).

**Comment by shinyquagsire23** [comment]
The music starting after the cantina is definitely intended behavior, the GOG release had CD audio emulation that failed to handle that behavior.

To be honest, I really need to build a period era Win9x PC just to figure out what the CD audio exactly did. It wouldn't surprise me if WinXP and later failed with MCI compatibility, or if some CD drives just didn't handle volume change requests. I can't really see any reason it would restart the music on a level restart though.

**Comment by darkstar252** [comment]
i have here a pentium 3 1000 MHz with windows 98se if hard disk cooperates, and a voodoo banshee, but i don't think there was a problem with windows xp or windows vista i try first on my pentium d 945 with windows vista , and i will see what is the behavior.

**Comment by Vortikai** [comment]
From the creator of jkgfxmod: "The TL;DR is that in 1997 CD audio worked in a really weird way. JK implemented its CD audio following doc, but Creative Labs didn't. Then when Microsoft reimplemented CD audio in software they emulated what Creative Labs did because it was the lowest common denominator."

It was a complex system of vying corporations. The "Disney" patched digital copies of JK and MotS are filled with janky hacky patches where COG verbs that control music volume work now and such, when they didn't for most people in 97.

This was a from a conversation spurred by me making modded music that adds simulated dynamic silence between tracks for JK, as I get ear fatigued easily when music is constantly on loop. The spaced out music with 1-2 minutes of silence between tracks is how I played JK and MOTS back in the day due to the audio bugs. Now it's all patched up, I had to "reimplement" the bug for my own preference :D
